File: python/src/temperature_conversion.py
class TemperatureConversion:
  def celsius_to_fahrenheit(self, celsius):
    return round((celsius * 9) / 5 + 32)

  def celsius_to_kelvin(self, celsius):
    return round(celsius + 273.15)

  def celsius_to_rankine(self, celsius):
    return round((celsius * 9) / 5 + 491.67)

  def fahrenheit_to_celsius(self, fahrenheit):
    return round(((fahrenheit - 32) * 5) / 9)

  def fahrenheit_to_kelvin(self, fahrenheit):
    return round(((fahrenheit - 32) * 5) / 9 + 273.15)

  def kelvin_to_celsius(self, kelvin):
    return round(kelvin - 273.15)

  def kelvin_to_fahrenheit(self, kelvin):
    return round(((kelvin - 273.15) * 9) / 5 + 32)

File: javascript/src/temperature-conversion.js
class TemperatureConversion {
  celsiusToFahrenheit = (celsius) => {
    return Math.round((celsius * 9) / 5 + 32);
  };

  celsiusToKelvin = (celsius) => {
    return Math.round(celsius + 273.15);
  };

  celsiusToRankine = (celsius) => {
    return Math.round((celsius * 9) / 5 + 491.67);
  };

  fahrenheitToCelsius = (fahrenheit) => {
    return Math.round(((fahrenheit - 32) * 5) / 9);
  };

  fahrenheitToKelvin = (fahrenheit) => {
    return Math.round(((fahrenheit - 32) * 5) / 9 + 273.15);
  };

  kelvinToCelsius = (kelvin) => {
    return Math.round(kelvin - 273.15);
  };

  kelvinToFahrenheit = (kelvin) => {
    return Math.round(((kelvin - 273.15) * 9) / 5 + 32);
  };
}

export default TemperatureConversion;
